go/exocom/src/exocom/exocom.go

```diff
@@ -95,16 +100,40 @@ func (e *ExoCom) listenForErrors() {
 	}
 }
 
+func (e *ExoCom) listenSender(message structs.Message) {
+	if e.routingManager.CanSend(message.Sender, message.Name) {
+		err := e.send(message)
+		if err != nil {
+			printLogError(e.logger.Error(err.Error()))
+		}
+	} else {
+		printLogError(e.logger.Warning(fmt.Sprintf("Warning: Service '%s' is not allowed to broadcast the message '%s'", message.Sender, message.Name)))
+	}
+}
+
 func (e *ExoCom) listenForMessages() {
 	for {
 		message := <-e.messageChannel
-		if e.routingManager.CanSend(message.Sender, message.Name) {
-			err := e.send(message)
-			if err != nil {
-				printLogError(e.logger.Error(err.Error()))
-			}
+		if !e.hasSecurity {
+			e.listenSender(message)
 		} else {
-			printLogError(e.logger.Warning(fmt.Sprintf("Warning: Service '%s' is not allowed to broadcast the message '%s'", message.Sender, message.Name)))
+			switch message.Name {
```

Comment:
Okay now that we have this logic working, can we try to extract it to a security manager struct? I imagine it could look something like:
```go
securityResult := e.securityManager.ReceiveMessage(message)
if securityResult.MessageToSend != nil {
  //...
} 
if securityResult.WarningMessageToLog != "" {
  //...
}
```
Where the securityManager could have a switch statement similar to the current one and encapsulates most of the logic that was added into exocom currently.

For the default case, the messageToSend is the serviceMessage created in the `sendToSecurity` method. It could then be passed to `e.send` since the security service is the only one that receives messages with the name "authorize message".